src/serialization/binary_archive.h

```diff
@@ -145,9 +141,11 @@ struct binary_archive<false> : public binary_archive_base<std::istream, false>
   template <class T>
   void serialize_uvarint(T &v)
   {
-    typedef std::istreambuf_iterator<char> it;
-    if (tools::read_varint(it(stream_), it(), v) < 0)
-      stream_.setstate(std::ios_base::failbit);
+    auto current = bytes_.begin();
+    auto end = bytes_.end();
+    good_ &= (0 <= tools::read_varint(current, end, v));
+    current = std::min(current, bytes_.end());
+    bytes_ = {current, std::size_t(bytes_.end() - current)};
```

Comment:
Might lead to integer overflow if we assume that `bytes_.end()` might be less than `current` (previous line 147).

Reply:
Nevermind, Missed the `current =` assignment.